code/game/gamemodes/nuclear/nuclear.dm

```diff
@@ -131,16 +132,36 @@ proc/issyndicate(mob/living/M as mob)
 		spawnpos++
 		update_synd_icons_added(synd_mind)
 
-	//update_all_synd_icons()
-
-	if(uplinklocker)
-		new /obj/structure/closet/syndicate/nuclear(uplinklocker.loc)
+	scale_telecrystals()
+	share_telecrystals()
 	if(nuke_spawn && synd_spawn.len > 0)
 		var/obj/machinery/nuclearbomb/syndicate/the_bomb = new /obj/machinery/nuclearbomb/syndicate(nuke_spawn.loc)
 		the_bomb.r_code = nuke_code
 
 	return ..()
 
+/datum/game_mode/nuclear/proc/scale_telecrystals()
+	var/danger
+	danger = GLOB.player_list.len
+	while(!IsMultiple(++danger,10))//Round up to the nearest multiple of ten
+	total_tc += danger * NUKESCALINGMODIFIER

```

Comment:
Is this supposed to be within your whole loop?

Reply:
No, the loop is one line only and ups the danger var to the next multiple of 10. That was taken from the original uplink management code, I'll make it nicer.